app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -49,9 +54,12 @@ class ManageIQ::Providers::Openstack::CloudManager < ManageIQ::Providers::CloudM
   supports :swift_service
   supports :create_host_aggregate
 
+  # TODO: move to after_initialization
   before_create :ensure_managers
 
+  # TODO: move to before_validation
   before_update :ensure_managers_zone_and_provider_region
```

Comment:
One thing to keep in mind, the tenant_id is set also in a `before_validation` callback so we need to make sure those play nicely together.  Don't want to be setting the child.tenant_id = tenant_id before tenant_id has been set

https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/tenancy_mixin.rb#L5